manifests/init.pp

```diff
@@ -260,7 +269,29 @@
     validate_absolute_path($keys_real)
   }
 
-  validate_string($restrict_options)
+  if is_array($restrict_options) == true {
+    $restrict_options_real = $restrict_options
+  }
+  # needed for backward compatibility
+  elsif is_string($restrict_options) == true {
+    $restrict_options_real = $restrict_options ? {
+      'USE_DEFAULTS' => $default_restrict_options,
+      default        => [ "-4 $restrict_options", "-6 $restrict_options", ]
+    }
+  }
+  else {
+    fail("restrict_options must be an array (prefered) or a string (will be deprecated).")
```

Comment:
technicaly restrict_options and restrict_localhost could get merged into one (would break backward compatibility now)